taskipy/task.py

```diff
@@ -0,0 +1,75 @@
+from typing import Optional
+
+from taskipy.exceptions import InvalidRunnerTypeError, TaskNotFoundError
+from taskipy.pyproject import PyProject
+
+
+class Task:
+    def __init__(self, task_name: str):
+        """
+        Sets up the task by saving it's name and
+        retrieiving the command it references in the
+        pyproject file.
+
+        Args:
+            task_name (str): The name of the task in [tools.taskipy.tasks].
+
+        Raises:
+            TaskNotFoundError: If the task could not be found in the
+                               [tools.taskipy.tasks] section.
+        """
+        self.project = PyProject()
```

Comment:
Perhaps the project should be loaded as part of the `TaskRunner` class. The config is an attribute of the program itself, not of the task, and therefore should be loaded as part of the runner's algorithm. This is 

That said, the transformation of task_name -> cmd should be done here; So let me suggest that we pass the configuration as part of the constructor into the class, rather than load it inline.

Reply:
in addition, as with other classes, let's make all class variables private :)

Reply:
Makes sense, I think this is the conclusion I reached above after reading your comments in the `TaskRunner` sketch before having read this.